src/main/kotlin/com/builzer/backend/global/util/JwtUtil.kt

```diff
@@ -0,0 +1,91 @@
+package com.builzer.backend.global.util
+
+import com.builzer.backend.global.config.security.JwtAuthenticationToken
+import com.builzer.backend.global.config.security.MemberPayload
+import io.jsonwebtoken.*
+import io.jsonwebtoken.security.Keys
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.security.core.Authentication
+import org.springframework.stereotype.Component
+import java.security.Key
+import java.time.Instant
+import java.util.*
+
+@Component
+class JwtUtil {
+    @Value("\${jwt.expiration.access}")
+    private lateinit var accessExpiration: String
+
+    @Value("\${jwt.expiration.refresh}")
+    private lateinit var refreshExpiration: String
+
+    private val key: Key = Keys.secretKeyFor(SignatureAlgorithm.HS512)
+
+    fun generateAccessToken(email: String, memberId: Long, gitAccessToken: String): String {
+        return Jwts.builder()
+            .setIssuedAt(Date.from(Instant.now()))
+            .setExpiration(expiration(accessExpiration))
+            .addClaims(claims(email = email, memberId = memberId, gitAccessToken = gitAccessToken))
+            .signWith(key)
+            .compact()
+    }
+
+    fun generateRefreshToken(email: String, memberId: Long, gitAccessToken: String): String {
+        return Jwts.builder()
+            .setIssuedAt(Date.from(Instant.now()))
+            .setExpiration(expiration(refreshExpiration))
+            .addClaims(claims(email = email, memberId = memberId, gitAccessToken = gitAccessToken))
+            .signWith(key)
+            .compact()
+    }
+
+    @Throws(
+        ExpiredJwtException::class,
+        UnsupportedJwtException::class,
+        MalformedJwtException::class,
+        SignatureException::class,
+        IllegalArgumentException::class
+    )
+    fun parseClaims(bearerToken: String): Claims {
+        val token = bearerToken.replace("Bearer ", "")
+        return Jwts.parserBuilder()
+            .setSigningKey(key)
+            .build()
+            .parseClaimsJws(token)
+            .body
+    }
+
+    fun getAuthentication(claims: Claims): Authentication {
+        return JwtAuthenticationToken(
+            memberPayload = MemberPayload(
+                id = extractMemberId(claims),
+                email = extractEmail(claims),
+                gitAccessToken = extractGitAccessToken(claims)
+            )
+        )
+    }
+
+    private fun extractEmail(claims: Claims): String {
+        return claims["email"] as String
```

Comment:
(should) 여기서도 클래스 강제 캐스팅보다는 if(claims["email"] is String) 으로 빼고 캐스팅이 안된다면 오류를 뱉는게 좋을 것 같아요 밑에 추출 클래스들도 마찬가지로요

Reply:
claims에는 Map<String, Object> 형식이 들어가야 해서, claims의 값을 꺼내 쓸 때는 강제 캐스팅을 해야 합니다.
```suggestion
        if (claims["email"] is String) {
            return claims["email"] // Type mismatch. Required: String, Found: Any?
        }
```